src/posix/platform/netif.cpp

```diff
@@ -1027,11 +1032,20 @@ static void processNetifInfoEvent(otInstance *aInstance, struct rt_msghdr *rtm)
 
 #endif
 
+#pragma GCC diagnostic push
+#pragma GCC diagnostic ignored "-Wcast-align"
+
 static void processNetlinkEvent(otInstance *aInstance)
 {
     const size_t kMaxNetifEvent = 8192;
     ssize_t      length;
-    char         buffer[kMaxNetifEvent];
+
+#if defined(__linux__)
+    alignas(nlmsghdr)
+#else
+    alignas(rt_msghdr)
+#endif
+        char buffer[kMaxNetifEvent];
```

Comment:
Is this alignment done by `clang-format`?

Reply:
Yes.